BLUESPAWN-client/src/monitor/Event.cpp

```diff
@@ -0,0 +1,187 @@
+#include "monitor/Event.h"
+#include "hunt/reaction/Log.h"
+#include "util/eventlogs/EventLogs.h"
+
+Event::Event(EventType type) : type(type) {}
+
+void Event::AddCallback(const std::function<void()>& callback) {
+	callbacks.push_back(callback);
+}
+
+void Event::RunCallbacks() const {
+	for(auto callback : callbacks){
+		callback();
+	}
+}
+
+/************************
+***   EventLogEvent   ***
+*************************/
+EventLogEvent::EventLogEvent(const std::wstring& channel, int eventID, const std::vector<EventLogs::XpathQuery>& queries) :
+	Event(EventType::EventLog), 
+    channel(channel), 
+	eventID(eventID), 
+	eventLogTrigger{ [this](EventLogs::EventLogItem){ this->RunCallbacks(); } } {}
+
+bool EventLogEvent::Subscribe(){
+	LOG_VERBOSE(1, L"Subscribing to EventLog " << channel << L" for Event ID " << eventID);
+	DWORD status{};
+	auto subscription = EventLogs::SubscribeToEvent(GetChannel(), GetEventID(), eventLogTrigger, queries);
+	if(subscription){
+		eventSub = *subscription;
+	}
+	return status;
+}
+
+std::wstring EventLogEvent::GetChannel() const {
+	return channel;
+}
+
+int EventLogEvent::GetEventID() const {
+	return eventID;
+}
+
+std::vector<EventLogs::XpathQuery> EventLogEvent::GetQueries() const {
+	return this->queries;
+}
+
+bool EventLogEvent::operator==(const Event& e) const {
+	if(e.type == EventType::EventLog && dynamic_cast<const EventLogEvent*>(&e)){
+		auto evt = dynamic_cast<const EventLogEvent*>(&e);
+		return evt->GetChannel() == channel && evt->GetEventID() == eventID;
+	} else return false;
+}
+
+/************************
+***   RegistryEvent   ***
+*************************/
+void RegistryEvent::DispatchRegistryThread(){
+	std::optional<RegistryEvent>* RegistryEvents = new std::optional<RegistryEvent>[MAXIMUM_WAIT_OBJECTS - 1]{};
+	RegistryEvent::hListener = CreateEventW(nullptr, false, false, nullptr);
+	RegistryEvents[0] = *RegistryEvent::subscribe;
+	RegistryEvent::subscribe = std::nullopt;
+	RegistryEventThreadArgs ThreadArgs = { *RegistryEvent::hListener, RegistryEvents };
+	HandleWrapper thread = CreateThread(nullptr, 0, reinterpret_cast<LPTHREAD_START_ROUTINE>(RegistryEvent::RegistryEventThreadFunction), &ThreadArgs, 0, nullptr);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'thread' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5092780)